framework/util/file_output_stream.cpp

```diff
@@ -59,10 +59,25 @@ FileOutputStream::~FileOutputStream()
     }
 }
 
+void FileOutputStream::Reset(FILE* file)
+{
+    if ((file_ != nullptr) && own_file_)
```

Comment:
Do we need some kind of warning here?  Also, what if `file` and `file_` are equal?

Reply:
In response to this I started making a patch that asserted that file and .file_ were not equal and did a log in release mode but when I was stepping through it I realised it was firing every time on Windows. An alternating sequence of `fopen` `fclose` seems to return the same FILE* every time. I could make a more complex test accounting for the `owned_` flag but I'm worried about writing something complex and confusing looking and potentially subtly wrong and then walking away.